The windows logic in std/path breaks in DNT

**Describe the bug**

I want to publish [a package](https://github.com/ventojs/vento) to NPM using DNT. It worked fine until now. After updating `@std/path` from 0.225.2 to 1.0.0, the build process breaks:

```
[dnt] Building project...
[dnt] Type checking ESM...
src/deps/jsr.io/@std/path/1.0.0/_os.ts:25:6 - error TS2304: Cannot find name 'navigator'.

25     (navigator.userAgent.includes("Win") ? "windows" : "linux");
        ~~~~~~~~~

error: Uncaught (in promise) Error: Had 1 diagnostics.
          throw new Error(`Had ${diagnostics.length} diagnostics.`);
                ^
    at getProgramAndMaybeTypeCheck (https://jsr.io/@deno/dnt/0.41.2/mod.ts:468:17)
    at build (https://jsr.io/@deno/dnt/0.41.2/mod.ts:354:17)
    at eventLoopTick (ext:core/01_core.js:168:7)
    at async file:///Users/oscarotero/www/ventojs/vento/_scripts/build_npm.ts:12:1
```

**Steps to Reproduce**

1. Clone the repository https://github.com/ventojs/vento
2. Run `deno task build-npm 1.12.10`

**Expected behavior**

The build is done successfully

**Environment**

- OS: MacOS 12.7.5
- deno version: 1.45.2
- std version: @std/path@1.0.0

--- a/path/_os.ts
+++ b/path/_os.ts
@@ -20,7 +20,10 @@ type OSType =
 function getOsType(): OSType {
   // deno-lint-ignore no-explicit-any
   return (globalThis as any).Deno?.build.os ||
-    (navigator.userAgent.includes("Win") ? "windows" : "linux");
+    // deno-lint-ignore no-explicit-any
+    ((globalThis as any).navigator?.userAgent.includes("Win")
+      ? "windows"
+      : "linux");
 }
 
 export const isWindows: boolean = getOsType() === "windows";

